room/room-compiler/src/main/kotlin/androidx/room/processor/DeletionMethodProcessor.kt

```diff
@@ -34,6 +34,12 @@ class DeletionMethodProcessor(
 
         val returnType = delegate.extractReturnType()
 
+        context.checker.check(
+            !delegate.isUsingMultipleConcurrencyPatterns(),
+            executableElement,
+            ProcessorErrors.USING_MULTIPLE_CONCURRENCY_PATTERNS
+        )
```

Comment:
You can move this and all the other `check` calls into the `ShortcutMethodProcessor` since they all need it.
Before returning from `extractReturnType()`.

```
// In ShortcutMethodProcessor
fun extractReturnType(): XType {
    context.checker.check(
        ...
    )
    return delegate.extractReturnType()
}
```

Reply:
Done